### PR TITLE
Shorten image importer dialTimeout to 5 seconds

### PR DESCRIPTION
**Pull request by liggitt**
Related to https://github.com/openshift/origin/issues/6381
Related to https://github.com/openshift/origin/issues/6259

### CODE REVIEW

**Comment by liggitt** [comment]
[test]

**Comment by liggitt** [comment]
@deads2k @stevekuznetsov PTAL
Once the parallel image import goes in, I would consider lengthening this to 10 seconds

**Comment by liggitt** [comment]
green jenkins... mysql:latest import was much better (~10 seconds)

this was still longer than I'd like, but I expect @deads2k's changes (parallel and abandoning NotFound imports) will significantly improve this:
```
SUCCESS after 42.423s: hack/../test/cmd/newapp.sh:143: executing 'oc get imagestreamtags installable:file' expecting success; re-trying every 0.2s until completion or 60.000s
```

**Comment by liggitt** [comment]
Merging to help the queue. Would still like review and can adjust in a follow up if needed

**Comment by liggitt** [comment]
[merge]

**Comment by liggitt** [comment]
Fun... flakes blocking flake fixes - https://github.com/openshift/origin/issues/6065
```
I1219 14:52:03.014415   21047 etcd.go:37] Deleting &etcd.Node{Key:"/kubernetes.io", Value:"", Dir:true, Expiration:(*time.Time)(nil), TTL:0, Nodes:etcd.Nodes(nil), ModifiedIndex:0x348, CreatedIndex:0x348} (child of &etcd.Node{Key:"", Value:"", Dir:true, Expiration:(*time.Time)(nil), TTL:0, Nodes:etcd.Nodes{(*etcd.Node)(0xc2086b8060), (*etcd.Node)(0xc2086b80c0)}, ModifiedIndex:0x0, CreatedIndex:0x0})
I1219 14:52:07.248341   21047 etcd.go:37] Deleting &etcd.Node{Key:"/openshift.io", Value:"", Dir:true, Expiration:(*time.Time)(nil), TTL:0, Nodes:etcd.Nodes(nil), ModifiedIndex:0x34f, CreatedIndex:0x34f} (child of &etcd.Node{Key:"", Value:"", Dir:true, Expiration:(*time.Time)(nil), TTL:0, Nodes:etcd.Nodes{(*etcd.Node)(0xc2086b8060), (*etcd.Node)(0xc2086b80c0)}, ModifiedIndex:0x0, CreatedIndex:0x0})
F1219 14:52:14.653034   21047 etcd.go:39] Unable to delete key: 100: Key not found (/openshift.io) [1124]
```
[merge]

**Comment by liggitt** [comment]
```
test-cmd.sh
Logging to /tmp/openshift-cmd/MPVw/logs...
openshift: openshift v1.1-582-g0add922 kubernetes v1.1.0-origin-1107-g4c8e6f4 etcd 2.1.2
[INFO] Create certificates for the OpenShift server to /tmp/openshift-cmd/MPVw/openshift.local.config/master
Generated new key pair as /tmp/openshift-cmd/MPVw/openshift.local.config/master/serviceaccounts.public.key and /tmp/openshift-cmd/MPVw/openshift.local.config/master/serviceaccounts.private.key
Generating node credentials ...
Created node config for 172.18.4.199 in /tmp/openshift-cmd/MPVw/openshift.local.config/node-172.18.4.199
Wrote master config to: /tmp/openshift-cmd/MPVw/openshift.local.config/master/master-config.yaml
Generating node credentials ...
Created node config for 172.18.4.199 in /tmp/openshift-cmd/MPVw/openshift.local.config/node-172.18.4.199
validation: ok
ERROR: gave up waiting for https://127.0.0.1:28443/healthz
```
[merge]

File: test/integration/dockerregistryclient_test.go [hunk 1]
```diff
@@ -63,7 +63,7 @@ func retryWhenUnreachable(t *testing.T, f func() error) error {
 }
 
 func TestRegistryClientConnect(t *testing.T) {
-	c := dockerregistry.NewClient()
+	c := dockerregistry.NewClient(10 * time.Second)
```
Comment (deads2k) [inline test/integration/dockerregistryclient_test.go]: Can I get some constants in the update?  This looks a little silly.  `DefaultTimeout` or something.  Even a separate constructor for the 90% who don't care.

**Pull request merged**